index.js

```diff
@@ -56,15 +50,17 @@ function fastifyWebsocket (fastify, opts, next) {
         throw new Error('invalid wsHandler function')
       }
 
-      router.on('GET', routeOptions.path, (req, _, params) => {
-        const result = wsHandler.call(fastify, req[kWs], req, params)
-
-        if (result && typeof result.catch === 'function') {
-          result.catch(err => req[kWs].destroy(err))
+      routeOptions.handler = (request, reply) => {
+        if (request.raw[kWs]) {
+          reply.hijack()
```

Comment:
Could you think of a reason why `hijack` would ever need to be async or return something? I can't off the top of my head, but this style of API that is super side-effect-ful always makes me a bit nervous.